shared/product.go

```diff
@@ -111,3 +112,26 @@ func CertRotate(product string, ips []string) (string, error) {
 
 	return "", nil
 }
+
+func SecretEncryptOps(action, ip, product string) (string, error) {
+	secretEncryptCmd := map[string]string{
+		"status":      fmt.Sprintf("sudo %s secrets-encrypt status", product),
+		"enable":      fmt.Sprintf("sudo %s secrets-encrypt enable", product),
+		"disable":     fmt.Sprintf("sudo %s secrets-encrypt disable", product),
+		"prepare":     fmt.Sprintf("sudo %s secrets-encrypt prepare", product),
+		"rotate":      fmt.Sprintf("sudo %s secrets-encrypt rotate", product),
+		"reencrypt":   fmt.Sprintf("sudo %s secrets-encrypt reencrypt", product),
+		"rotate-keys": fmt.Sprintf("sudo %s secrets-encrypt rotate-keys", product),
+	}
+
+	stdout, err := RunCommandOnNode(secretEncryptCmd[action], ip)
+
+	if err != nil {
+		return ip, ReturnLogError(fmt.Sprintf("FATAL: secrets-encryption %s action failed", action), err)
```

Comment:
why returning IP on error ? are you using it even on error flow ?

Reply:
fixed. returning ""